src/server/http/short_url_error.js

```diff
@@ -1,9 +1,8 @@
 import Boom from 'boom';
 
-export function handleShortUrlError(err) {
-  if (err.isBoom) return err;
-  if (err.status === 401) return Boom.unauthorized();
-  if (err.status === 403) return Boom.forbidden();
-  if (err.status === 404) return Boom.notFound();
-  return Boom.badImplementation();
+export function handleShortUrlError(error) {
+  if (!error.isBoom && !(error instanceof Error)) {
```

Comment:
We really should not support non-errors here. Pretty sure this was spurred by the tests failing because they [aren't sending real errors](https://github.com/elastic/kibana/blob/master/src/server/http/__tests__/short_url_error.js#L6-L12), which means that this code isn't actually testing what would happen in production with this condition. I think this check should go, we should always use `Boom.wrap()` and the tests should use real error objects.